src/forms/redux-form/selectInput.test.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import ReduxFormSelectInput from './SelectInput';
+
+describe('redux-form SelectInput', () => {
+	const testOptions = [
+		{ label: 'One', value: '1' },
+		{ label: 'Two', value: '2' },
+		{ label: 'Three', value: '3' }
+	];
+	const formAttrs = {
+		label: 'Countries',
+		name: 'formSelectCountries',
+		options: testOptions,
+		required: true,
+		meta: {
+			touched: false,
+			error: 'Did you mean Batman and Robin?'
+		}
+	};
+
+	it('renders a SelectInput component with expected attributes from mock data', () => {
+		const component = shallow(<ReduxFormSelectInput {...formAttrs} />);
+		expect(component).toMatchSnapshot();
+	});
+
+	it('does not render error if field is not touched and validateBeforeTouched is true', () => {
+		const props = {
+			...formAttrs,
+			validateBeforeTouched: true,
+			meta: {
+				...formAttrs.meta,
+				touched: false
+			}
+		};
+		const component = mount(<ReduxFormSelectInput {...props} />);
+		expect(component.find('.field--error').exists()).toBe(false);
```

Comment:
is `field--error` a constant somewhere? or is it `text--error`?

Reply:
We have both `field--error` and `text--error`, which both seem to be standard error stylings. I think `field--error` is for the component and `text--error` is for the error text itself.

Reply:
@unjust - `field-error` is a thing. We use it on other inputs